packages/system-access/src/base/credentials.ts

```diff
@@ -0,0 +1,105 @@
+import type { Logger } from '@sap-ux/logger';
+import { getService, BackendSystemKey, BackendSystem } from '@sap-ux/store';
+import type { UrlAbapTarget } from '../types';
+import { isAppStudio } from '@sap-ux/btp-utils';
+import prompts from 'prompts';
+import { questions } from './prompts';
+
+export type BasicAuth = Required<Pick<BackendSystem, 'username' | 'password'>>;
+export type ServiceAuth = Required<Pick<BackendSystem, 'serviceKeys' | 'name'>> & { refreshToken?: string };
+
+/**
+ * Check the secure storage if it has credentials for the given target.
+ *
+ * @param target ABAP target
+ * @param logger - reference to the logger instance
+ * @returns credentials from the store or undefined.
+ */
+export async function getCredentialsFromStore<T extends BasicAuth | ServiceAuth | undefined>(
+    target: UrlAbapTarget,
+    logger: Logger
+): Promise<T | undefined> {
+    try {
+        if (!isAppStudio()) {
+            const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+            let system = await systemService.read(new BackendSystemKey({ url: target.url, client: target.client }));
+            // check if there are credentials for the default client
+            if (!system && target.client) {
+                system = await systemService.read(new BackendSystemKey({ url: target.url }));
+            }
+            return system as T;
+        }
+    } catch (error) {
+        logger.warn('Reading credentials from store failed');
+        logger.debug(error.message);
+    }
+    return undefined;
+}
+
+/**
+ * Store the credentials in the secure storage.
+ *
+ * @param name system name
+ * @param target target
+ * @param target.url system url
+ * @param target.client optional system client
+ * @param credentials basic auth credentials
+ * @param credentials.username username
+ * @param credentials.password password
+ * @param logger reference to the logger instance
+ * @returns true if the credentials are successfully stored
+ */
+export async function storeCredentials(
+    name: string,
+    target: { url: string; client?: string },
+    credentials: { username: string; password: string },
+    logger: Logger
+): Promise<boolean> {
+    try {
+        const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+        const system = new BackendSystem({
+            name,
+            ...target,
+            ...credentials
+        });
+        await systemService.write(system);
+        return true;
+    } catch (error) {
+        logger.error('Could not store credentials.');
+        logger.debug(error);
+        return false;
+    }
+}
+
+/**
+ * Checks the environment variables for Fiori tools settings.
+ *
+ * @returns basic auth credentials from the environment or undefined.
+ */
+export function getCredentialsFromEnvVariables(): BasicAuth | undefined {
```

Comment:
I don't see this function used in any productive coding. Only in unit tests.

Reply:
I wanted to use it here: https://github.com/SAP/open-ux-tools/blob/main/packages/backend-proxy-middleware/src/base/proxy.ts#L338, so that the usage of the `env` variables `FIORI_TOOL_*` are all in one module.

Reply:
OK, sounds good! 👍

Reply:
I didn't do it and would also prefer keeping it for yet another PR because using the new `system-access` module in the `backend-proxy-middleware` will lead to more (good) enhancements of the `system-access` e.g. support for destination instances or corporate proxies. Adding this here as well would make it harder to review.